### DISCUSSION TITLE
What would happen if the "data" directory was leaked? Would this compromise the password vault?

### DISCUSSION THREAD

**Discussion by yuhr123** [dani-garcia/vaultwarden#4595] [Q&A] [answered]
Hi everyone, 

I deployed Vaultwarden on the cloud using Docker and mounted the host machine's "data" directory to the container through bind mounting.

I noticed that there are SQLite databases, RSA public and private keys in the "data" directory. Knowing these belong to my password vault, I'm curious if a hacker who gained access to this server would be able to compromise all of my passwords?

<img width="441" alt="image" src="https://github.com/dani-garcia/vaultwarden/assets/3388604/be960aa2-a0f7-4d4c-9965-e62d34589455">

**Comment by BlackDex** [comment, marked as answer]
Short answer, No.

The RSA Key is only used to generate and validate JWT's which are used for login/invite to validate a session/login.
Those are not used to encrypt/decrypt any password data.

Almost all data in the database is encrypted except for items which need to be decrypted like data for your account like email and name, or data for the organization like the name. All other items are encrypted by the clients and always sent encrypted over the wire.

Same goes for attachments/sends, that data is encrypted by the clients and send fully encrypted to the server.
See https://bitwarden.com/help/what-encryption-is-used/ on how Bitwarden encrypts the data.

**Reply by yuhr123** [reply]
Thank you for your answer. 😉
